src/components/PrimaryPanes/SourcesTree.js

```diff
@@ -184,6 +184,8 @@ class SourcesTree extends Component<Props, State> {
     const { sources } = this.props;
     const blackBoxedPart =
       item.contents.get &&
+      item.contents.has("id") &&
+      sources.has(item.contents.get("id")) &&
       sources.get(item.contents.get("id")).get("isBlackBoxed")
```

Comment:
This might be split into two parts to reduce the time it takes to parse.

Reply:
Done